lms/templates/header/user_dropdown.html

```diff
@@ -17,9 +17,9 @@
 ## This template should not use the target student's details when masquerading, see TNL-4895
 self.real_user = getattr(user, 'real_user', user)
 profile_image_url = get_profile_image_urls_for_user(self.real_user)['medium']
-username = self.real_user.username
+profile_display_name = self.real_user.profile.display_name
```

Comment:
We should use `display_name_with_default` in place of `display_name` throughout.